### PR TITLE
Adds Elasticsearch / X-Pack configuration via Env variables.

### PR DESCRIPTION
**Pull request by marcinc**

### CODE REVIEW

File: kibana.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@
 # is proxied through the Kibana server.
 # elasticsearch.username: "user"
 # elasticsearch.password: "pass"
+%ELASTICSEARCH_USERNAME%
```
Comment (vaijab) [inline kibana.yml]: Should this read `elasticsearch.username: %ELASTICSEARCH_USERNAME%` ?

File: run.sh [hunk 2]
```diff
@@ -2,9 +2,39 @@
 
 export HOME=/kibana
 
+: ${ELASTICSEARCH_URL:=http://localhost:9200}
 : ${ELASTICSEARCH_CERTIFICATE_VERIFICATION:=true}
+: ${ELASTICSEARCH_SSL_CA_CERT_PATH:=}
+: ${ELASTICSEARCH_USERNAME:=}
+: ${ELASTICSEARCH_PASSWORD:=}
+: ${XPACK_SECURITY_ENABLE:=false}
+: ${XPACK_SECURITY_ENCRYPTION_KEY:=$(uuidgen)}
```
Comment (vaijab) [inline run.sh:11]: Is it OK to regen this every time kibana runs?

**Review by vaijab — CHANGES_REQUESTED** [review]

File: run.sh (hunk 2 above)
Comment (marcinc) [reply run.sh:11]: Yes. It's an arbitrary string of 32 characters or more used to encrypt credentials in a cookie. If it changes the worse that can happen is that user will be asked to sign in again.

File: kibana.yml (hunk 1 above)
Comment (marcinc) [reply kibana.yml]: This is a bit ugly I agree but with default empty values for both username and password this seem to break configuration. The workaround for that is to include entire line only if username / password is specified.

Comment (jits) [reply kibana.yml]: Should probably use a different name then, given the user specified env var is also called `ELASTICSEARCH_USERNAME`?

**Review by jits — APPROVED** [review]
One minor thing, otherwise LG.

File: kibana.yml (hunk 1 above)
Comment (marcinc) [reply kibana.yml]: @jits This is our convention. `ELASTICSEARCH_USERNAME` env variable replaces a placeholder of the same name `%ELASTICSEARCH_USERNAME%`.

Comment (jits) [reply kibana.yml]: But in this case you're not directly replacing the placeholder `%ELASTICSEARCH_USERNAME%` with the value of the env var `ELASTICSEARCH_USERNAME` – you're deriving a new output from the original env var which gets put here, as Vaidas pointed out above.

Comment (jits) [reply kibana.yml]: (General point being it could be clearer when someone looks at it in a few months time :D)

Comment (marcinc) [reply kibana.yml]: Yes, in this case if env var is specified it'll basically add entire config line with it's value placed there. I can change the placeholder name if that helps?

**Comment by marcinc** [comment]
@jits Changed the placeholder names. Hope that's more clear now?

**Comment by jits** [comment]
@marcinc – looks great to me!

File: run.sh (hunk 2 above)
Comment (vaijab) [reply run.sh:11]: How does it work when there is more than one pod and a user hits different pod each request?

Comment (marcinc) [reply run.sh:11]: @vaijab One can always pass a fixed value in environment to ensure all instances use the same encryption key.

**Review by vaijab — APPROVED** [review]

**Pull request merged**